core/src/main/java/org/mskcc/oncotree/utils/TumorTypesUtil.java

```diff
@@ -20,32 +26,22 @@
 /**
  * Created by Hongxin on 2/25/16.
  */
+@Component
 public class TumorTypesUtil {
-    private static final String PROPERTY_FILE = "classpath:application.properties";
-    private static List<String> TumorTypeKeys = Arrays.asList("code", "name", "nci", "level", "umls", "maintype", "color");
 
-    public static Map<String, TumorType> getTumorTypes() {
-        Map<String, TumorType> tumorTypes = new HashMap<>();
-        Properties properties = getProperties();
-        try {
-            Version version = VersionUtil.getVersion("realtime");
-            CacheUtil.resetMainTypesByVersion(version);
+    private final static Logger logger = Logger.getLogger(TumorTypesUtil.class);
 
-            tumorTypes = parseFromRaw(new FileInputStream(properties.getProperty("tumor_type_file_path")), version);
-        } catch (FileNotFoundException e) {
-            e.printStackTrace();
-        }
-        return tumorTypes;
-    }
+    private static OncoTreeRepository oncoTreeRepository;
+    @Autowired
+    public void setOncoTreeRepository(OncoTreeRepository property) { oncoTreeRepository = property; }
```

Comment:
I guess this will happen on instantiation of the class. I thought that an Autowired class member (putting @Autowired before the member declaration for oncoTreeRepository) would accomplish something similar. But I guess the set method might have additional logic that you want to happen.